be/btree.c

```diff
@@ -10137,102 +10276,969 @@ static int btree_ut_thread_init(struct btree_ut_thread_info *ti)
 {
 	int rc;
 
-	M0_ALLOC_ARR(ti->ti_rnd_state_ptr, 64);
-	if (ti->ti_rnd_state_ptr == NULL) {
-		return -ENOMEM;
-	}
+	M0_ALLOC_ARR(ti->ti_rnd_state_ptr, 64);
+	if (ti->ti_rnd_state_ptr == NULL) {
+		return -ENOMEM;
+	}
+
+	M0_SET0(&ti->ti_random_buf);
+	initstate_r(ti->ti_rng_seed_base, ti->ti_rnd_state_ptr, 64,
+		    &ti->ti_random_buf);
+
+	srandom_r(ti->ti_thread_id + 1, &ti->ti_random_buf);
+
+	rc = m0_thread_confine(&ti->ti_q, &ti->ti_cpu_map);
+
+	m0_bitmap_fini(&ti->ti_cpu_map);
+	return rc;
+}
+
+#define FILL_KEY_MTREE_UT(key, ksize, data)                                    \
+	do {                                                                   \
+		uint32_t i;                                                    \
+		key[0] = data;                                                 \
```

Comment:
Try to ensure key is of type (uint32_t *) by writing the following line:
(void)(&i == &key[0]);

Reply:
Ok.

Reply:
Done